commit 121a6a17439b000b9699c3fa876636db20fa4107
Author: Gerd Hoffmann <kraxel@redhat.com>
Date:   Fri Oct 11 10:01:09 2013 +0200

    drm/cirrus: use drm_set_preferred_mode
    
    Explicitly set 1024x768 as default mode, so the display doesn't come up
    with the largest supported mode.
    
    While being at it drop first three drm_add_modes_noedid calls.  As
    drm_add_modes_noedid fills the mode list with modes from the database
    *up to* the specified size it is pretty pointless to call it multiple
    times with different sizes.
    
    Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
    Signed-off-by: Dave Airlie <airlied@redhat.com>

--- a/drivers/gpu/drm/cirrus/cirrus_mode.c
+++ b/drivers/gpu/drm/cirrus/cirrus_mode.c
@@ -494,13 +494,12 @@ static struct drm_encoder *cirrus_encode
 
 int cirrus_vga_get_modes(struct drm_connector *connector)
 {
-	/* Just add a static list of modes */
-	drm_add_modes_noedid(connector, 640, 480);
-	drm_add_modes_noedid(connector, 800, 600);
-	drm_add_modes_noedid(connector, 1024, 768);
-	drm_add_modes_noedid(connector, 1280, 1024);
+	int count;
 
-	return 4;
+	/* Just add a static list of modes */
+	count = drm_add_modes_noedid(connector, 1280, 1024);
+	drm_set_preferred_mode(connector, 1024, 768);
+	return count;
 }
 
 static int cirrus_vga_mode_valid(struct drm_connector *connector,
